src/fswAlgorithms/attGuidance/sunSearch/sunSearch.h

```diff
@@ -0,0 +1,63 @@
+/*
+ ISC License
+
+ Copyright (c) 2024, Laboratory for Atmospheric and Space Physics, University of Colorado at Boulder
+
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+ */
+
+#ifndef _SUN_SEARCH_
+#define _SUN_SEARCH_
+
+#include "architecture/_GeneralModuleFiles/sys_model.h"
+#include "architecture/utilities/bskLogging.h"
+#include "architecture/messaging/messaging.h"
+#include "architecture/msgPayloadDefC/NavAttMsgPayload.h"
+#include "architecture/msgPayloadDefC/VehicleConfigMsgPayload.h"
+#include "architecture/msgPayloadDefC/AttGuidMsgPayload.h"
+#include "cMsgCInterface/AttGuidMsg_C.h"
+
+
+/*! @brief A class to perform EMA SEP pointing */
+class SunSearch: public SysModel {
+public:
+    SunSearch();
+    ~SunSearch();
+    void SelfInit();                                             //!< Self initialization for C-wrapped messages
+    void Reset(uint64_t CurrentSimNanos);
+    void UpdateState(uint64_t CurrentSimNanos);
+
+    ReadFunctor<NavAttMsgPayload>          attNavInMsg;          //!< input msg measured attitude
+    ReadFunctor<VehicleConfigMsgPayload>   vehConfigInMsg;       //!< input veh config msg
+    Message<AttGuidMsgPayload>             attGuidOutMsg;        //!< Attitude reference output message
+    AttGuidMsg_C                           attGuidOutMsgC = {};  //!< C-wrapped attitude guidance output message
+
+private:
+    double    slewTime[3];          //!< [s] total time for the three-axes maneuver - user-defined
```

Comment:
:large_orange_diamond: What are your thoughts on refactoring each of these variables into a struct. That way the various methods that use them will not have to access each value with `index` everywhere. Happy to chat.

Reply:
I've combined all the properties into one big structure. Much cleaner overall.
I left out the inertias and the maximum torques because they pertain to each principal body axis, and not necessarily to the requested slews.

Reply:
Solid.